staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -251,6 +252,26 @@ func (s *Validator) validateExpressions(ctx context.Context, fldPath *field.Path
 			continue
 		}
 		if evalResult != types.True {
+			tryMessage := true
+			if len(rule.MessageExpression) != 0 {
+				evalResult, evalDetails, err := compiled.MessageExpression.ContextEval(ctx, activation)
+				if evalDetails == nil {
+					klog.Warningf("runtime cost could not be calculated for message expression %q")
+				}
+				if err != nil {
+					klog.Warningf("error evaluating message expression, falling back: %v", err.Error())
+				} else {
+					messageStr, ok := evalResult.Value().(string)
+					if !ok {
+						klog.Warningf("could not could message expression result to string, falling back: %v", evalResult.Value())
+					}
+					errs = append(errs, field.Invalid(fldPath, sts.Type, messageStr))
+					tryMessage = false
+				}
+			}
+			if !tryMessage {
+				continue
+			}
```

Comment:
Is `tryMessage` needed? Is it better than invoking`continue` at the same place where `tryMessage` is set to `false`?

Reply:
I've refactored the function a bit, so now `tryMessage` is gone.